test/test_controller.js

```diff
@@ -0,0 +1,364 @@
+var Token = artifacts.require("MFC_Token");
+var Controller = artifacts.require("ICO_controller");
+var Web3 = require('web3');
+
+
+Web3.providers.HttpProvider.prototype.sendAsync = Web3.providers.HttpProvider.prototype.send
+var web3 = new Web3(new Web3.providers.HttpProvider("http://localhost:8545"));
+
+
+
+function wait(delay) {
+    var stop = new Date().getTime() / 1000;
+    while (new Date().getTime() / 1000 < stop + delay) {
```

Comment:
For greater clarity, better:
var stop = new Date ().getTime () / 1000 + delay;
or
var start = new Date ().getTime () / 1000;